scio-core/src/main/scala/com/spotify/scio/util/TupleFunctions.scala

```diff
@@ -35,5 +35,7 @@ private[scio] object TupleFunctions {
     (kv.getKey, kv.getValue.asScala)
 
   def kvListToTuple[K, V](kv: KV[K, JList[V]]): (K, Iterable[V]) =
-    (kv.getKey, kv.getValue.asInstanceOf[JIterable[V]].asScala)
+    (kv.getKey, new Iterable[V] {
+      override def iterator: Iterator[V] = kv.getValue.iterator().asScala
```

Comment:
🎉 glad to see that this idea works!

Reply:
What's the different? Something wrong with `JIterableWrapper`?

Reply:
Because `JIterableWrapper` is a case class in Scala's collections, it is using case classes's equals which means that a `JIterableWrapper` is not equal to an Iterable defined by `Iterable(...)`

Example:
```
scala> Arrays.asList(1, 2, 3).asInstanceOf[java.lang.Iterable[Int]].asScala.toIterable == Iterable(1,2,3)
res18: Boolean = false
```

This behavior is hacked around in the tests when we have only one element iterables, by using a helper class:
```
def iterable[T](elems: T*): Iterable[T] = {
    val l = new java.util.ArrayList[T]()
    elems.foreach(l.add)
    l.asInstanceOf[JIterable[T]].asScala
  }
```
and then the tests are:
```
r1 should
        containInAnyOrder(Seq(("a", iterable(1)), ("b", iterable(12)), ("c", iterable(23))))
```

This causes confusion and gives rise to a very non-idiomatic way of expressing one element iterables.

Example test case:
```
it should "not confuse users" in {
    runWithContext{
      sc =>
        import scala.collection.JavaConverters._
        val p = sc.parallelize(Seq(("a", 1)))
        // The following fails
        p.topByKey(1).values should containInAnyOrder(Seq(Iterable(1))) 
        // The following works
        p.topByKey(1).values should containInAnyOrder(Seq(java.util.Arrays.asList(1).asInstanceOf[java.lang.Iterable[Int]].asScala.toIterable))
    }
  }
```

Reply:
Heh right it's been so long I almost forgot having to do that.